odl/solvers/functional/default_functionals.py

```diff
@@ -2193,6 +2198,169 @@ def __repr__(self):
+
+    .. math::
+        F(x)
+        =
+        \\begin{cases}
+            0 & \\text{if } x_i \geq 0 \\forall i \\text{ and } \sum_i x_i = r
+            \\\\
+            +\\infty & \\text{else.}
+        \\end{cases}
+
+    where `r` is the radius.
+    """
+
+    def __init__(self, space, radius=1):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `DiscreteLp` or `FnBase`
+            Domain of the functional.
+        radius : positive float, optional
+            Radius `r` of simplex.
+        """
+        super().__init__(space=space, linear=False, grad_lipschitz=np.nan)
+        self.radius = radius
+
+    def __call__(self, x):
+        """Return whether x is in the unit simplex."""
+
+        r = self.radius
+        if (x.ufuncs.sum() == r and x.ufuncs.greater_equal(0).asarray().all()):
+            return 0
+
+        return np.inf
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional. The indicator functional is not
+        differentiable over the entire domain.
+        """
+
+        raise NotImplementedError('No yet implemented')
+
+    @property
+    def proximal(self):
+        """Return the `proximal factory` of the functional."""
+
+        domain = self.domain
+
+        class ProximalSimplex(Operator):
+            """Proximal operator implemented by the algorithm of [1].
+
+            [1] Duchi, John, et al. "Efficient projections onto the l 1-ball
+            for learning in high dimensions." Proceedings of the 25th
+            international conference on Machine learning. ACM, 2008.
+            """
+
+            def __init__(self, sigma):
+                self.sigma = sigma
+                super().__init__(domain=domain, range=domain, linear=False)
+
+            def _call(self, x, out):
+
+                # projection onto simplex
+                proj_simplex(x, self.radius, out)
+
+        return ProximalSimplex
+
+    @property
+    def convex_conj(self):
+        """The convex conjugate."""
+
+        raise NotImplementedError('Not yet implemented')
+
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        return '{}({!r})'.format(self.__class__.__name__, self.domain)
+
+
+class IndicatorSum(Functional):
+
+    """The indicator functional of a unit sum constraint.
+
+    Notes
+    -----
+    The functional :math:`F` is given by:
+
+    .. math::
+        F(x)
+        =
+        \\begin{cases}
+            0 & \\text{if } \sum_i x_i = 1 \\\\
+            +\\infty & \\text{else.}
+        \\end{cases}
+    """
+
+    def __init__(self, space):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `DiscreteLp` or `FnBase`
+            Domain of the functional.
+        """
+        super().__init__(space=space, linear=False, grad_lipschitz=np.nan)
+
+    def __call__(self, x):
+        """Return whether x has unit sum."""
+
+        if x.ufuncs.sum() == 1:
```

Comment:
With this implementation the value will practically always be infinity. A problem? Maybe consider allowing some tolerance.

Reply:
Good idea. How do we achieve that? It is also not clear how much tolerance to allow.

Reply:
You could use the built-in `round` function: `round(x.ufuncs.sum(), ndigits=6) == 1`. 

The tolerance level is a bit arbitrary, agreed, but since we're comparing against 1, the error is also "normalized". So any value that "works" for practical purposes (i.e., single precision arithmetic) should be okay.

As a guidance, consider the following experiment:
```python
>>> x = np.random.uniform(0, 1, size=int(1e9))
>>> y = x.astype(np.float32)
>>> xsum = x.sum()
>>> x /= xsum
>>> y /= xsum
>>> abs(1 - x.sum())
1.5543122344752192e-15
>>> abs(1 - y.sum())
8.106231689453125e-06
```
This is a relatively large example, but it shows the order of error that you'd have to expect from *stable* computations. So maybe it makes sense to define the required precision based on the data type.